c/datasetjson.c

```diff
@@ -2251,9 +2254,238 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
+    snprintf(field, sizeof(field)-1, "STORCLAS(%s) ", request->storageClass);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize file data type */
+  if (request->fileData != NULL) {
+    snprintf(field, sizeof(field)-1, "FILEDATA(%s) ", request->fileData);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize file data type */
+  if (request->eattr != NULL) {
+    strncpy (tempString, request->eattr, sizeof (tempString) -1);
+    strupcase(tempString); 
+    if (strncmp( tempString, "TRUE", 5)) {
+      snprintf(field, sizeof(field)-1, "EATTR(OPT) ");
+      strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+    }
+  }
+
+  /* Initialize number of generations */
+  if (request->numGenerations != 0) {
+    snprintf(field, sizeof(field)-1, "MAXGENS(%d) ", request->numGenerations);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+ 
+  /* Initialize volumn if defined */
+  if (request->volume != NULL) {
+    snprintf(field, sizeof(field)-1, "VOL(%s) ", request->volume);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize dataClass if defined */
+  if (request->dataClass != NULL) {
+    snprintf(field, sizeof(field)-1, "DATACLAS(%s) ", request->dataClass);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize management class if defined */
+  if (request->manageClass != NULL) {
+    snprintf(field, sizeof(field)-1, "MGMTCLAS(%s) ", request->manageClass);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize management class if defined */
+  if (request->averageRecord != NULL) {
+    //snprintf(field, sizeof(field)-1, "BLOCK(%s) ", request->averageRecord);
+    snprintf(field, sizeof(field)-1, "AVGREC(%s) ", request->averageRecord);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize expiration data  */
+  if (request->expiration != NULL) {
+    snprintf(field, sizeof(field)-1, "EXPDL(%s) ", request->expiration);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize expiration data  */
+  if (request->spaceUnit != NULL) {
+    snprintf(field, sizeof(field)-1, "SPACE(%s) ", request->spaceUnit);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /****************************/
+  /* Create the qsam data set */
+  /****************************/
+  printf (" REQUEST: %s\n",  below2G->request);
```

Comment:
Naked printf.

Reply:
This was left in for debugging.  It has been removed.